elrond.md

```diff
@@ -205,13 +210,312 @@ Parallelized semantics can be achieved by instead using the following rule:
 
 Here, host calls are implemented, by defining the semantics when `hostCall(MODULE_NAME, EXPORT_NAME, TYPE)` is left on top of the `instrs` cell.
 
+#### Completing the Call
+
+```k
+    rule <instrs> hostCall("env", "finish", [ i32 i32 .ValTypes ] -> [ .ValTypes ]) => local.get 0 ~> local.get 1 ~> #finish ... </instrs>
+
+    syntax EndCall ::= "#finish"
+ // ----------------------------
+    rule <instrs> #finish => #done ... </instrs>
+         <valstack> <i32> LENGTH : <i32> OFFSET : VS => VS </valstack>
+         <callee> CALLEE </callee>
+         <returnData> _ => #getBytesRange(DATA, OFFSET, LENGTH) </returnData>
+         <returnStatus> _ => Finish </returnStatus>
+         <account>
+           <address> CALLEE </address>
+           <code> MODIDX:Int </code>
+           ...
+         </account>
+         <moduleInst>
+           <modIdx> MODIDX </modIdx>
+           <memAddrs> 0 |-> MEMADDR </memAddrs>
+           ...
+         </moduleInst>
+         <memInst>
+           <mAddr> MEMADDR </mAddr>
+           <msize> SIZE </msize>
+           <mdata> DATA </mdata>
+           ...
+         </memInst>
+         requires (OFFSET +Int LENGTH) <=Int (SIZE *Int #pageSize())
+
+
+    syntax Done ::= "#done"
+ // -----------------------
+    rule <instrs> #done ~> _REST => . </instrs>
+```
+
+#### Call State
+
+```k
+    rule <instrs> hostCall("env", "getNumArguments", [ .ValTypes ] -> [ i32 .ValTypes ]) => i32.const size(ARGS) ... </instrs>
+         <callingArguments> ARGS </callingArguments>
+
+    rule <instrs> hostCall("env", "getArgumentLength", [ i32 .ValTypes ] -> [ i32 .ValTypes ]) => i32.const lengthArg({ARGS[IDX]}:>Argument) ... </instrs>
+         <locals> 0 |-> <i32> IDX </locals>
+         <callingArguments> ARGS </callingArguments>
```

Comment:
Will crash if IDX >= length of ARGS, not important now, but we should keep in mind

Reply:
You're right, let's guard this with a requires-clause.